common/matchmaking.ts

```diff
@@ -936,3 +936,21 @@ export interface AddMatchmakingSeasonResponse {
 export interface GetCurrentMatchmakingSeasonResponse {
   season: MatchmakingSeasonJson
 }
+
+export type MatchmakingTypeAndSeasonId = Tagged<
```

Comment:
I don't think you really need this to be tagged? Doesn't the string typing stuff in TS just handle this properly now?

(I would prefer this type doesn't exist here, this is an implementation detail of the things that do this, not something we should be making into a common type we use a ton of different places)

Reply:
Hmm, if I define a map like this
```ts
const map = new Map<`${MatchmakingType}|${SeasonId}`, number>()
```

And then try to use it like this:
```ts
const key = `${matchmakingType}|${seasonId}`
const value = map.get(key)
```

I get the following error:

![image](https://github.com/user-attachments/assets/5eb95512-fe44-4dda-b47d-cb08196e4b3e)

But if I use it like this:
```ts
const value = map.get(`${matchmakingType}|${seasonId}`)
```

Then it works fine. Weird.

Reply:
Would guess this is just a limitation of type inference, and if you explicitly defined the type of `key` it would work the other way too. Or just wait a few typescript versions and maybe they'll fix this :P